includes/search.rs

```diff
@@ -738,76 +738,122 @@ pub(crate) fn filter_for_operator(
     operator: PredicateOperator,
 ) -> Result<(), String> {
     for object in unfiltered {
-        let val = match search_field {
-            "id" => &object[search_field],
+        let mut val = match search_field {
+            "id" => object[search_field].clone(),
             _ => {
                 let data = &object["data"];
-                &data[search_field]
+                data[search_field].clone()
             }
         };
 
+        let ids = &object["ids"];
+        println!(
+            "(filter_for_operator) (MOCK) test ids field [{}].",
+            ids,
+        );
+        if val == serde_json::json!(null) && *ids != serde_json::json!(null) {
+            println!(
+                "(filter_for_operator) (MOCK) found ids [{}].",
+                ids,
+            );
+            let ids: Vec<super::FieldValue> = serde_json::from_value(ids.clone()).map_err(|e| format!("Could not convert [{:?}] to Ids from json value: {}", ids, e))?;
+            println!("search ids after convert: {:?}", ids);
+            for id in ids {
+                if id.field == search_field {
+                    val = serde_json::json!(id.value)
+                }
+            }
+        }
+
+        println!("(filter_for_operator) (MOCK) got value [{}] for object [{}].", val, object);
         match operator {
             PredicateOperator::Equals => {
                 let search_val: String = get_single_search_value(search_values)?;
-                let val = val.to_string();
-                log::debug!(
-                    "(filter_for_operator) Equals filter with value [{}] for val [{}].",
+                let cmp_val: String;
+                if let Some(string) = val.as_str() {
```

Comment:
This section confuses me a little bit, is the output of [`serde_json::json!`](https://docs.rs/serde_json/latest/serde_json/macro.json.html) somehow simultaneously a String and an option type? How can it be passed to format!() in the else case?

Additionally the json! macro has panic behavior, would prefer serde_json::to_string and serde_json::from_string with error handling if possible

Reply:
fixed: [chore: code style fixes](https://github.com/aetheric-oss/svc-storage/pull/105/commits/d05d63638158b993e616bd1cea105add3ec43535)